### ISSUE TITLE
Does not run in Windows 10

### ISSUE THREAD

**Issue by verdy-p** [BiglySoftware/BiglyBT#4]
Windows 10 does not let us run the installer, apparently it is deteted as violating some security rules.
Several antimalwares are blocking it, including Avast, and Windows Defender.
Avast performs a scan and blocks it preventively, waiting for an online analysis, there's a button to say "yes I want to use this file", but this choice is not remembered, and the installer still refuses to run.
Apparently this is caused by the fact that the small installer actually does not contain the software but attempts to download it from an unsecure website.
Do you have an installer that performs offline installation, which is fully self contained (not requiring any internet connection) ? For now this installer for Windows does not work at all.

**Comment by TuxPaper** [comment]
Try the installer from our GitHub release page https://github.com/BiglySoftware/BiglyBT/releases and let us know if that works.

The installer on our website is a stub installer that detects whether you are 32 or 64 bit, and downloads the correct full installer (from https, so it's odd it would report it's unsecure).   The GitHub release page has the full installer, but you have to know whether you are 32 or 64 bit (most likely 64 bit)

**Comment by trgarner** [comment]
I installed from their homepage, installed worked on windows 10 w/ defender running, no errors.  It did take some time to port over my vuze settings.  But all seems to be working as advertised at this time.

**Comment by Havokdan** [comment]
My kaspersky blocks the installer stub, but no issues with those that are in Github.

**Comment by DeathStalker77** [comment]
Not the place (yet) I know - but hoping you will be getting a forum up and running soon.  Very anxious for 2.0's release  :)

**Comment by parg** [comment]
Do you have any suggestions for good, free forum software or services? :) We have used https://mybb.com/ in the past but it is quite a bit of work to maintain and also it appears to be undergoing a major rewrite for 'version 2' that I'm not sure I want to be a tester for!

**Comment by verdy-p** [comment]
Effectively it was blocked by several antivirus, but now after they checked
it online, the installer will accept to run.
It was blocked due to its uncommon use of networking options and
installation mode.
Ideally, you should avoid the issue by providing full installers (all
contained in one file) instead of just an online installer (downloading the
components from unsecured hosts)
If you use secure hosts, you need a correct, non-expired certificate, not
just for the hosting domain, but also for signing the software components
themselves: as it is a java application, you normally need to include
manifests in your zip (if you include binary components such as a few DLLs,
that must also themselves be signed and should have a secure API if it is
used to bypass some OS restrictions normally applied with the standard Java
API), and in your JAR's (containing compiled java classes).
Not doing this will cause these alerts.
So effectively the software installer was denied running, as it was highly
suspect given the APIs it uses which should be restricted and secured, or
if you include some code performing some decryption for its own code, or if
it uses some known third party libraries that are commonly reused by many
malwares, for example for networking, or to bypass NAT restrictions, or
change the default TCP/IP behavior, or if you use "raw" sockets).
Anyway this software (just like Azureus or ) still has problems connecting
through common NATs implemented in home routers provided by ISPs: its UPNP
implementation does not work correctly (no such problem with µTorrent or
BitTorrent): incoming connections are extremely unstable and long to be
detected, and the connection graphs are also extremely slow to be
constructed. Even for downloading the few samples from your site is
EXTREMELY slow: lot of difficulties to find peers, and once connected,
extremely small bandwidth used.
For now the software is just unusable, and cannot replace µTorrent.

2017-08-04 3:52 GMT+02:00 TuxPaper <notifications@github.com>:

**Comment by verdy-p** [comment]
Put the links to the full installers in 32-bit and 64-bit on your page.
Visibly there's a problem when running the light installer to let it
download the actual installers, even if they are on an HTTPS site on GitHub.

2017-08-04 3:52 GMT+02:00 TuxPaper <notifications@github.com>:

**Comment by DeathStalker77** [comment]
@parg - I have not done so, but I am certainly on enough forums that I can ask around as to what is the easiest  :)  REALLY hoping to see this one succeed.  I'm not a coder, but I am a developer/designer/UX.

**Comment by TuxPaper** [comment]
I've put the 32 bit and 64 bit installers on the download page.  I've also changed the default download from the stub to 64 bit if the browser indicates it's 64 bit.  Finally, I built a new stub installer which will give it a different id and hopefully stop triggering those detectors.

Closing issue for now.  A new issue can be started if there's similar issues in the future.

**Comment by csylvain** [comment]
AVG+Win10 held up the install process because the executable was novel (not in their database, apparently) but I got underway again by telling AVG to stand down and let the executable run. All good afterward.
